packages/nodejs/src/cli/diagnose.ts

```diff
@@ -0,0 +1,249 @@
+    console.log(`  Installation result`)
+    console.log(`    Status: success`)
+    console.log(`  Language details`)
+    console.log(`    Node.js version: ${data["host"]["language_version"]}`)
+    console.log(`  Download details`)
+    console.log(
+      `    Download URL: ${data["installation"]["download"]["download_url"]}`
+    )
+    console.log(`    Checksum: ${data["installation"]["download"]["checksum"]}`)
+    console.log(`  Build details`)
+    console.log(`    Install time: ${data["installation"]["build"]["time"]}`)
+    console.log(
+      `    Architecture: ${data["installation"]["build"]["architecture"]}`
+    )
+    console.log(`    Target: ${data["installation"]["build"]["target"]}`)
+    console.log(
+      `    Musl override: ${data["installation"]["build"]["musl_override"]}`
+    )
+    console.log(
+      `    Linux ARM override: ${data["installation"]["build"]["linux_arm_override"]}`
+    )
+    console.log(
+      `    Library type: ${data["installation"]["build"]["library_type"]}`
+    )
+    console.log(`  Host details`)
+    console.log(`    Root user: ${data["installation"]["host"]["root_user"]}`)
+    console.log(
+      `    Dependencies: ${this.format_value(
+        data["installation"]["host"]["dependencies"]
+      )}`
+    )
+
+    this.print_newline()
+
+    console.log(`  Host information`)
+    console.log(`    Architecture: ${data["host"]["architecture"]}`)
+    console.log(`    Operating System: ${data["host"]["os"]}`)
+    console.log(`    Node.js version: ${data["host"]["language_version"]}`)
+    console.log(`    Root user: ${this.format_value(data["host"]["root"])}`)
+    console.log(
+      `    Running in container: ${this.format_value(
+        data["host"]["running_in_container"]
+      )}`
+    )
+
+    this.print_newline()
+
+    console.log(`Configuration`)
+    console.log(
+      `  Environment: ${this.format_value(data["config"]["options"]["env"])}`
+    )
+
+    console.log(
+      `  debug: ${this.format_value(data["config"]["options"]["debug"])}`
+    )
+
+    console.log(`  log: ${this.format_value(data["config"]["options"]["log"])}`)
+
+    console.log(
+      `  endpoint: ${this.format_value(data["config"]["options"]["endpoint"])}`
+    )
+    console.log(
+      `  ca_file_path: ${this.format_value(
+        data["config"]["options"]["ca_file_path"]
+      )}`
+    )
+    console.log(
+      `  active: ${this.format_value(data["config"]["options"]["active"])}`
+    )
+    console.log(
+      `  push_api_key: ${this.format_value(
+        data["config"]["options"]["push_api_key"]
+      )}`
+    )
+
+    this.print_newline()
+
+    console.log(`Read more about how the diagnose config output is rendered`)
+    console.log(`https://docs.appsignal.com/nodejs/command-line/diagnose.html`)
+
+    this.print_newline()
+
+    console.log(`Validation`)
+    console.log(
+      `  Validating Push API key: ${this.colorize(
+        data["validation"]["push_api_key"]
```

Comment:
Not sure if it's in scope but I noticed that this output isn't accurate.

Given this config output (with an undefined push_api_key, which I also don't know if we should print it as `undefined`):

```
Configuration
  ...
  push_api_key: undefined
```

I get this validation:

```
Validation
  Validating Push API key: valid
```

See also the admin view of this report: https://appsignal.com/admin/diagnose_reports/60e41eb35ac13f2d186f711f

Reply:
That should be caused by https://github.com/appsignal/appsignal-nodejs/issues/408, which causes the result to return “valid” for any input. I’ve created a separate issue to fix that to make sure this moves along.